source/iopipe/json/serialize.d

```diff
@@ -57,77 +48,95 @@ struct DefaultDeserializationPolicy {
         ref bool[N] visited
     ) {
         static assert(N == SerializableMembers!T.length);
-        alias members = SerializableMembers!T;
-        alias ignoredMembers = AllIgnoredMembers!T;
-        // Check each member to see if it matches
-        switch(key)
-        {
-            static foreach(i, memberName; members) {
-                { // Add a block scope to contain each declaration, avoiding duplicate jsonName
-                  // declarations in the foreach loop.
-                  static if(hasUDA!(__traits(getMember, T, memberName), alternateName)) {
-                      enum jsonName = getUDAs!(__traits(getMember, T, memberName), alternateName)[0].name;
-                  }
-                  else {
-                      enum jsonName = memberName;
-                  }
-                  
-                  case jsonName:
-                      // Choose appropriate deserialization method based on member type               
-                      deserializeItem(this, tokenizer, __traits(getMember, item, memberName));
-                      visited[i] = true; // Mark as visited
-                      
-                      if(this.relPol == ReleasePolicy.afterMembers)
-                        tokenizer.releaseParsed();
-                      return;
-                  
-                }
-            }
-
-            static if(ignoredMembers.length > 0)
-            {
-                static foreach(m; ignoredMembers)
-                {
-                    static foreach(s; m.ignoredMembers)
-                    {
-                    case s:
-                    }
-                }
-                // ignored members are ignored if they show up
-                tokenizer.skipItem();
-                break;
-            }
-
-            default:
-            // If we get here, it's truly an unknown field
-            throw new JSONIopipeException(format("No member named '%s' in type `%s`", key, T.stringof));
+        .onField(this, tokenizer, item, key, visited);
+        if(relPol == ReleasePolicy.afterMembers) {
+            tokenizer.releaseParsed();
         }
     }
     
-
-    
     // Called at end of deserialization
     void onObjectEnd(JT, T, size_t N)(ref JT tokenizer, ref T item, ref bool[N] visited) {
```

Comment:
I'm not sure this function is needed either.

I think the only code in here is to release the data after a member. Which is already being done in the onField function (even for skipped fields).

I believe the onField call is the only one that is needed to implement the release policy.